solidity/contracts/SisypheanExchange.sol

```diff
@@ -1,21 +1,259 @@
+	function returnRepBond(uint256 _universeId, uint256 _marketId) external {
+		Universe memory universe = universes[_universeId];
+		MarketResolutionData memory marketResolutionData = marketResolutions[_universeId][_marketId];
+		require(marketResolutionDataIsFinalized(marketResolutionData), "Cannot withdraw REP bond before finalized");
+
+		universe.reputationToken.transfer(marketResolutionData.initialReporter, REP_BOND);
+	}
+
+	function isFinalized(uint256 _universeId, uint256 _marketId) external view returns (bool) {
+		MarketResolutionData memory marketResolutionData = marketResolutions[_universeId][_marketId];
+		return marketResolutionDataIsFinalized(marketResolutionData);
+	}
+
+	function marketResolutionDataIsFinalized(MarketResolutionData memory marketResolutionData) internal view returns (bool) {
+		return marketResolutionData.reportTime != 0 && block.timestamp > marketResolutionData.reportTime + DISPUTE_PERIOD;
+	}
+
+	function getWinningOutcome(uint256 _universeId, uint256 _marketId) public view returns (uint256) {
+		MarketResolutionData memory marketResolutionData = marketResolutions[_universeId][_marketId];
+		require(marketResolutionDataIsFinalized(marketResolutionData), "Market is not finalized");
+
+		return marketResolutionData.outcome;
+	}
+
+	// TODO: Currently escalation game is a single dispute. Likely will be more complex.
+	function dispute(uint256 _universeId, uint256 _marketId, uint256 _outcome) external {
+		Universe memory universe = universes[_universeId];
+		require(universe.forkingMarket == 0, "Universe is forked");
+		MarketResolutionData memory marketResolutionData = marketResolutions[_universeId][_marketId];
+		require(block.timestamp < marketResolutionData.reportTime + DISPUTE_PERIOD, "Market not in dispute window");
+		require(_outcome < 3, "Invalid outcome");
+
+		universe.reputationToken.transferFrom(msg.sender, address(this), REP_BOND * 2);
+
+		for (uint256 i = 1; i < 4; i++) {
+			uint256 childUniverseId = (_universeId << 4) + i;
+			universes[childUniverseId] = Universe(
+				new ReputationToken(),
+				0,
+				0,
+				0,
+				universe.ethBalance
+			);
+		}
+
+		// TODO resolve each in respective universe
+
+		universe.forkingMarket = _marketId;
+		universe.forkTime = block.timestamp;
+		universes[_universeId] = universe;
+	}
+
+	function migrateREP(uint256 universeId, uint256 amount, uint256 outcome) external {
+		require(outcome < 3, "Invalid outcome");
+		Universe memory universe = universes[universeId];
+		require(block.timestamp < universe.forkTime + REP_MIGRATION_WINDOW, "Universe not in REP migration window");
+
+		uint256 softBurnedREP = universe.reputationToken.balanceOf(address(1));
+		uint256 correspondingETH = amount * universe.ethBalance / (universe.reputationToken.totalSupply() - softBurnedREP);
+
+		// Genesis is using REPv2 which we cannot actually burn
+		if (universeId == 0) {
+			universe.reputationToken.transferFrom(msg.sender, address(1), amount);
```

Comment:
I prefer `0xdead...` as it is much more obviously a burn address than `1` (which could just be a bug/typo).
```suggestion
			universe.reputationToken.transferFrom(msg.sender, address(0xdeaddeaddeaddeaddeaddeaddeaddeaddeaddead), amount);
```

That being said, I believe there are tools that treat transfer to `0` as a burn and transfer from `0` as a mint.  I dislike this, but if we care about third party tooling support for mint/burn it may matter.

Reply:
I think a _few_ things treat 0x1 as a designated burn address but its not as common as 0x0 which we can't do since the genesis REP (REPv2) wont allow it, so may as well go with `0xdead`